chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/G3Sed2015DownloadableResourceCreatorJob.java

```diff
@@ -0,0 +1,159 @@
+    private static final String EDITION = "2015";
+
+    @Autowired
+    private SchedulerCertifiedProductSearchDetailsAsync schedulerCertifiedProductSearchDetailsAsync;
+    
+    /**
+     * Default constructor.
+     * @throws Exception if issue with context
+     */
+    public G3Sed2015DownloadableResourceCreatorJob() throws Exception {
+        super();
+    }
+
+    @Override
+    public void execute(final JobExecutionContext jobContext) throws JobExecutionException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+        schedulerCertifiedProductSearchDetailsAsync.setLogger(LOGGER);
+
+        LOGGER.info("********* Starting the G3 SED 2015 Downloadable Resource Creator job. *********");
+        try {
+            List<Long> listingIds = getRelevantListingIds();
+
+            List<Future<CertifiedProductSearchDetails>> futures = getCertifiedProductSearchDetailsFutures(listingIds);
+            Map<Long, CertifiedProductSearchDetails> cpMap = getMapFromFutures(futures);
+            List<CertifiedProductSearchDetails> orderedListings = 
+                    createOrderedListOfCertifiedProducts(cpMap, listingIds);
+
+            File downloadFolder = getDownloadFolder();
+            writeToFile(downloadFolder, orderedListings);
+        } catch (Exception e) {
+            LOGGER.error(e);
+        }
+        LOGGER.info("********* Completed the G3 SED 2015 Downloadable Resource Creator job. *********");
+    }
+
+    /**
+     * Gets all listings that have certified to 170.315 (g)(3)
+     * @return
+     * @throws EntityRetrievalException
+     */
+    private List<Long> getRelevantListingIds() throws EntityRetrievalException {
+        LOGGER.info("Finding all listings attesting to " + CRITERIA_NAME + ".");
+        CertificationCriterionDTO certCrit = getCriteriaDao().getByNameAndYear(CRITERIA_NAME, EDITION);
+        List<Long> listingIds = getCertificationResultDao().getCpIdsByCriterionId(certCrit.getId());
+        LOGGER.info("Found " + listingIds.size() + " listings attesting to " + CRITERIA_NAME + ".");
+        return listingIds;
+    }
+
+    private List<Future<CertifiedProductSearchDetails>> getCertifiedProductSearchDetailsFutures(
+            final List<Long> listingIds) throws Exception {
+
+        List<Future<CertifiedProductSearchDetails>> futures = new ArrayList<Future<CertifiedProductSearchDetails>>();
+        SchedulerCertifiedProductSearchDetailsAsync cpsdAsync = getCertifiedProductDetailsAsyncRetrievalHelper();
+        
+        for (Long currListingId : listingIds) {
+            try {
+                futures.add(cpsdAsync.getCertifiedProductDetail(currListingId, getCpdManager()));
+            } catch (EntityRetrievalException e) {
+                LOGGER.error("Could not retrieve certified product details for id: " + currListingId, e);
+            }
+        }
+        return futures;
+    }
+
+    private Map<Long, CertifiedProductSearchDetails> getMapFromFutures(
```

Comment:
Would it make sense to move this (and maybe some other) function to a super class?